File: ui/libs/translatr-model/src/lib/model/oidc-provider-status.ts
export interface OidcProviderStatus {
  key: string;
  listed: boolean;
  active: boolean;
  provider: string | null;
  authServerUrl: string | null;
  clientId: string | null;
  /** Masked as "***len:<N>***", or null when no secret is configured. */
  clientSecret: string | null;
  scopes: string[];
  errors: string[];
}

File: src/main/java/com/translatr/service/OidcProviderStatus.java
package com.translatr.service;

import java.util.List;

/** Diagnostic view of one auth provider. {@code clientSecret} is already masked or null. */
public record OidcProviderStatus(
        String key,
        boolean listed,
        boolean active,
        String provider,
        String authServerUrl,
        String clientId,
        String clientSecret,
        List<String> scopes,
        List<String> errors) {
}
